File: main-server/src/main/kotlin/com/techtaurant/mainserver/blog/entity/Blog.kt
package com.techtaurant.mainserver.blog.entity

import com.techtaurant.mainserver.common.base.EntityBase
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Table
import org.hibernate.annotations.SQLDelete
import org.hibernate.annotations.SQLRestriction
import java.util.Date

/**
 * 블로그 엔티티
 *
 * Soft Delete 적용:
 * - 삭제 시 실제 DB에서 제거되지 않고 deleted_at에 현재 시간 기록
 * - 조회 시 deleted_at IS NULL인 데이터만 자동으로 필터링
 */
@Entity
@Table(name = "blogs")
@SQLDelete(sql = "UPDATE blogs SET deleted_at = CURRENT_TIMESTAMP WHERE id = ?")
@SQLRestriction("deleted_at IS NULL")
class Blog(

    @Column(nullable = false, length = 100, unique = true)
    var name: String,

    @Column(name = "display_name", length = 200)
    var displayName: String?,

    @Column(name = "icon_url", length = 500)
    var iconUrl: String?,

    @Column(name = "base_url", columnDefinition = "TEXT")
    var baseUrl: String?,

    @Column(name = "deleted_at")
    var deletedAt: Date? = null,
) : EntityBase()

File: techtaurant-crawler/app/domains/blog/entity/blog.py
"""
Blog 엔티티

블로그 정보를 저장하는 모델
"""

from __future__ import annotations

from datetime import datetime
from typing import TYPE_CHECKING
from uuid import UUID, uuid4

from sqlalchemy import DateTime, String, Text
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.configs.database import Base

if TYPE_CHECKING:
    from app.domains.article_link.entity.article_link import ArticleLink


class Blog(Base):
    """
    블로그 정보

    크롤링 대상 블로그의 기본 정보를 저장합니다.
    """

    __tablename__ = "blogs"

    id: Mapped[UUID] = mapped_column(primary_key=True, default=uuid4)
    name: Mapped[str] = mapped_column(String(100), nullable=False, unique=True, index=True)
    display_name: Mapped[str | None] = mapped_column(String(200), nullable=True)
    icon_url: Mapped[str | None] = mapped_column(String(500), nullable=True)
    base_url: Mapped[str | None] = mapped_column(Text, nullable=True)

    created_at: Mapped[datetime] = mapped_column(DateTime, nullable=False, default=datetime.utcnow)
    updated_at: Mapped[datetime] = mapped_column(
        DateTime, nullable=False, default=datetime.utcnow, onupdate=datetime.utcnow
    )
    deleted_at: Mapped[datetime | None] = mapped_column(DateTime, nullable=True, index=True)

    # Relationship
    article_links: Mapped[list[ArticleLink]] = relationship(
        "ArticleLink", back_populates="blog", cascade="all, delete-orphan"
    )
